accounting/src/delta/delta_data_collection/mod.rs

```diff
@@ -19,29 +19,28 @@ use self::undo::*;
 use std::collections::{btree_map::Entry, BTreeMap};
 
 use serialization::{Decode, Encode};
-use utils::ensure;
 
 use crate::error::Error;
 
-/// The outcome of combining two deltas for a given key upon the map that contains it.
-/// Every combine can produce either a new delta, that should be insert in the collection or a No-op
-/// in which case a delta is removed from the collection.
-enum DeltaMapOp<T> {
-    /// Insert a specific value (for example, to write a Create or Modify operation)
-    Insert(DeltaMapElement<T>),
-    /// Erase the value at the relevant key spot (for example, a Create followed by Delete yields nothing)
-    Erase,
-}
-
 /// Basic primitive that represent a difference introduced to the data.
 #[derive(PartialEq, Eq, Clone, Encode, Decode, Debug)]
 pub enum DataDelta<T> {
-    /// Stores new value
-    Create(T),
-    /// Stores previous and new values
-    Modify(T, T),
-    /// Stores value to restore
-    Delete(T),
+    // TODO: better error reporting?
+    Mismatch,
+    Modify(Option<T>, Option<T>),
+}
+
+impl<T: Clone> DataDelta<T> {
+    /// Returns an invert delta that has the opposite effect of the provided delta
+    /// and serves as an undo object
+    fn invert(&self) -> DataDeltaUndo<T> {
```

Comment:
invert should take self as argument to avoid unnecessary cloning. The caller should decide whether cloning is desirable.